Proposal for a process supporting on-boarding new team members

lirantal opened the pull request:
I would like to propose this document as a supporting process that helps on-board new members to the team.

The process also introduces the concept of a *buddy*, which we can further discuss about scope and responsibilities, but I think it can be helpful for newcomers.

Would be happy to get some feedback on any other items you find relevant to address in this process @dgonzalez @MarcinHoppe since you're both fresh on it.

vdeturckheim commented on processes/security_team_onboarding.md:
```diff
@@ -0,0 +1,39 @@
+This document aims to provide a clear process for on-boarding new team members to the Node.js Security WG.
+
+## Introducing new members
+New members may be nominated to join the WG, where they share an affinity to web or software security research, or activism around security and the Node.js project. All of which make good criteria to join the team but are not a closed list.
```
What about self-nomination?

MarcinHoppe replied:
It would also be good to clarify who is responsible for finding a buddy for a new WG member.

lirantal replied:
I think that should come from the WG team through the nomination/join process

vdeturckheim commented on processes/security_team_onboarding.md:
```diff
@@ -0,0 +1,39 @@
+
+
+### How to nominate new members?
+Open an issue in https://github.com/nodejs/security-wg that suggests to nominate a new team member. Provide as much background and context as possible which is relevant to this candidate and his potential future contributions to the Security WG.
```
I'd prefer a PR here (rather than waiting for :+1: on two diferent places)

MarcinHoppe commented on processes/security_team_onboarding.md:
```diff
@@ -0,0 +1,39 @@
+Please follow:
+* New member should open a PR with his username added to [security-wg/third_party_vuln_process.md at master · nodejs/security-wg · GitHub](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md#the-triage-team), where the username list is ordered alphabetically.
+* New members should familiarize themselves with the 3rd party vulnerability triage process at: [security-wg/third_party_vuln_process.md at master · nodejs/security-wg · GitHub](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md)
+* New members should schedule a demo session with their mentor to review the HackerOne platform where the new member receives a walk-through of our processes.
```
Is this the same session as the 30 minute touch-base call described in the "The Buddy Program" section or a separate one?

lirantal replied:
yeah. but time length isn't a hard-limit inn advance.

MarcinHoppe commented:
One other idea that I had: it would be nice to have a checklist issue with all of the items above that we would close when a new member is fully up to speed.

GitHub seems to allow for multiple issue templates now. I'll be happy to create a PR with the on-boarding issue template when we have the contents of this PR nailed down.

Also, should we have a document that describes how do members of the WG step down? I imagine there are a number things that would need to happen as well (revoking access, etc.)?

lirantal commented:
Both ideas are great @MarcinHoppe, PR 'em :-)

lirantal commented:
@vdeturckheim @MarcinHoppe updated.

mhdawson commented on processes/security_team_onboarding.md:
```diff
@@ -0,0 +1,39 @@
+
+A *buddy* is an active member of the WG who is familiar with the above topics, and should be recommended on the nomination issue/PR.
+
+A suggested on-boarding session for triage team members: 30 minutes touch-base call to cover the triage process on HackerOne and the repo.
```
This might be specific to people who will be involved in the triage of 3rd party vulnerabilities which is not necessarily everybody in the security-wg.

lirantal replied:
That's what I'm wondering - even if you're not in the triage team, wouldn't you want a quick tour of stuff we do, people involved, processes, etc? or would you say that all of that can be retrieved from the repo and our documents?

mhdawson commented on processes/security_team_onboarding.md:
```diff
@@ -0,0 +1,39 @@
+Please follow:
+* New member should open a PR with his username added to [security-wg/third_party_vuln_process.md at master · nodejs/security-wg · GitHub](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md#the-triage-team), where the username list is ordered alphabetically.
+* New members should familiarize themselves with the 3rd party vulnerability triage process at: [security-wg/third_party_vuln_process.md at master · nodejs/security-wg · GitHub](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md)
+* New members should schedule a demo session with their mentor to review the HackerOne platform where the new member receives a walk-through of our processes.
```
The reference to the HackerOne review probably fits better here.

MarcinHoppe commented:
@lirantal The HackerOne overview and walkthrough is priceless :). I have yet to see if PRing the vulnerability in this repo would be easier if done together.

lirantal commented:
Thanks @MarcinHoppe ;-)

Addressed comment from Michael to re-arrange triage demo session into the triage-specific on-boarding section.

Will merge later tonight if there aren't any other requests for changes.
